### PR TITLE
Add `switchToCorrectNetwork` method

### LINKED ISSUE #119: Switch to correct network
Whenever a user is on the wrong network, we will show a button to switch to the required network.
On clicking this button, the network will get changed to the required network instead of manually opening the MetaMask and changing the network.

### PR DESCRIPTION
**Pull request by Nazeeh21**
Closes #119 

## Description
This PR adds a method `switchToCorrectNetwork` to the `useWallet()` hook. This method allows the user to connect to the required network without manually heading over to the MetaMask and switching the network.

Users can use this method by directly assigning this `switchToCorrectNetwork` function to the `onClick` handler of any button and then the magic will happen and the MetaMask network will get changed to the required network.

In case, any user doesn't have MetaMask installed then it will show an alert to the user and ask them to install MetaMask.

### CODE REVIEW

**Comment by Dhaiwat10** [comment]
@Nazeeh21 can you check the failing test?

**Comment by Pponwili1998** [comment]
fix/issue-119

**Comment by Pponwili1998** [comment]
maim

File: packages/hooks/src/stories/ConnectWallet.stories.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
-import React, { useEffect } from 'react';
 import { Button } from '@chakra-ui/react';
-import { NETWORKS, Provider, useWallet } from '..';
-import Authereum from 'authereum';
 import Portis from '@portis/web3';
+import Authereum from 'authereum';
+import React, { useEffect } from 'react';
+import { NETWORKS, Provider, useWallet } from '..';
```
Comment (Dhaiwat10) [inline packages/hooks/src/stories/ConnectWallet.stories.tsx:5]: Why did you change the order of the imports? We should try to keep the changes to the minimum. Only introduce relevant changes so that it's easier for anyone to grasp the PR when re-visiting

File: packages/hooks/src/hooks/useWallet.ts [hunk 2]
```diff
@@ -22,16 +22,56 @@ export function useWallet() {
...
+            });
+          } catch (addError) {
+            console.error(addError);
+          }
+        }
+      }
+    } else {
+      // if no window.ethereum then MetaMask is not installed
+      alert(
+        'MetaMask is not installed. Please consider installing it: https://metamask.io/download.html'
```
Comment (Dhaiwat10) [inline packages/hooks/src/hooks/useWallet.ts:70]:
```suggestion
        'Switching networks automatically is only supported in MetaMask. https://metamask.io/download.html'
```

Can you change the error message?

File: packages/hooks/src/hooks/useWallet.ts [hunk 3]
```diff
@@ -22,16 +22,56 @@ export function useWallet() {
...
+        // check if the chain to connect to is installed
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: `0x${network}` }], // chainId must be in hexadecimal numbers
+        });
+      } catch (error) {
+        console.error(error);
+        // This error code indicates that the chain has not been added to MetaMask.
+        // If it is not, then install it into the user MetaMask
+        // @ts-ignore
```
Comment (Dhaiwat10) [inline packages/hooks/src/hooks/useWallet.ts:50]: Instead of `@ts-ignore`, did you try using something `@ts-expect-error`? I think that would be more appropriate.

File: packages/hooks/src/hooks/useWallet.ts [hunk 4]
```diff
@@ -22,16 +22,56 @@ export function useWallet() {
...
+        console.log('requiredNetwork', { network });
+        // check if the chain to connect to is installed
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: `0x${network}` }], // chainId must be in hexadecimal numbers
+        });
+      } catch (error) {
+        console.error(error);
+        // This error code indicates that the chain has not been added to MetaMask.
+        // If it is not, then install it into the user MetaMask
```
Comment (Dhaiwat10) [inline packages/hooks/src/hooks/useWallet.ts:49]:
```suggestion
        // If it is not, then install it into the user's MetaMask
```

File: packages/hooks/src/hooks/useWallet.ts [hunk 5]
```diff
@@ -22,16 +22,56 @@ export function useWallet() {
...
+                },
+              ],
+            });
+          } catch (addError) {
+            console.error(addError);
+          }
+        }
+      }
+    } else {
+      // if no window.ethereum then MetaMask is not installed
```
Comment (Dhaiwat10) [inline packages/hooks/src/hooks/useWallet.ts:68]:
```suggestion
      // if window.ethereum is undefined then MetaMask is not installed
```

**Review by Dhaiwat10 — CHANGES_REQUESTED** [review]

**Comment by Dhaiwat10** [comment]
You need to also run `yarn changeset` and do a minor bump to the `hooks` package

File: packages/hooks/src/stories/ConnectWallet.stories.tsx (hunk 1 above)
Comment (Nazeeh21) [reply packages/hooks/src/stories/ConnectWallet.stories.tsx:5]: There is not any specific reason for that. I'll change it back. 😅

File: packages/hooks/src/hooks/useWallet.ts (hunk 2 above)
Comment (Nazeeh21) [reply packages/hooks/src/hooks/useWallet.ts:70]: Sure, I'll update it.

File: packages/hooks/src/hooks/useWallet.ts (hunk 3 above)
Comment (Nazeeh21) [reply packages/hooks/src/hooks/useWallet.ts:50]: No, I haven't tried it, I'll try it.

**Review by Dhaiwat10 — APPROVED** [review]
